### PR TITLE
Add Haiku support

### PR DESCRIPTION
**Pull request by diversys**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -293,6 +293,9 @@ if(APPLE)
         target_link_libraries(keepassx_core "-framework LocalAuthentication")
     endif()
 endif()
+if(HAIKU)
+    target_link_libraries(keepassx_core network)
```
Comment (droidmonkey) [inline src/CMakeLists.txt]: What is this library?

File: CMakeLists.txt [hunk 2]
```diff
@@ -192,11 +192,13 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     add_gcc_compiler_flags("-Werror")
 endif()
 
+if (UNIX AND NOT HAIKU)
```
Comment (droidmonkey) [inline CMakeLists.txt]: This would break non-unix builds. Should simply be ```if(NOT HAIKU)```

File: src/CMakeLists.txt (hunk 1 above)
Comment (diversys) [reply src/CMakeLists.txt]: libnetwork.so is where network fictions live in Haiku.

Comment (droidmonkey) [reply src/CMakeLists.txt]: would you only want this if "WITH_XC_NETWORKING" is true?

File: share/CMakeLists.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@ file(GLOB DATABASE_ICONS icons/database/*.png)
 
 install(FILES ${DATABASE_ICONS} DESTINATION ${DATA_INSTALL_DIR}/icons/database)
 
-if(UNIX AND NOT APPLE)
+if(UNIX AND NOT APPLE AND NOT HAIKU)
```
Comment (droidmonkey) [inline share/CMakeLists.txt]: On second read this should actually be: ```if(UNIX AND NOT (APPLE OR HAIKU))```

File: src/autotype/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 if(WITH_XC_AUTOTYPE)
-    if(UNIX AND NOT APPLE)
+    if(UNIX AND NOT APPLE AND NOT HAIKU)
```
Comment (droidmonkey) [inline src/autotype/CMakeLists.txt]: Same here ```if(UNIX AND NOT (APPLE OR HAIKU))```

Comment (phoerious) [reply src/autotype/CMakeLists.txt]: That makes no sense. They are logically equivalent.

Comment (droidmonkey) [reply src/autotype/CMakeLists.txt]: Making this change allows MacOS to build again, so I disagree (or at least CMake does)

Comment (droidmonkey) [reply src/autotype/CMakeLists.txt]: nevermind, I jumped the gun. CI failed for some unknown reason

**Comment by droidmonkey** [comment]
Closed in favor of #2859. In the future, create a new branch when you create a PR. If you reuse develop or master I cannot rebase them properly.

**Comment by diversys** [comment]
Thanks!

**Pull request closed without merging**